10-gis.Rmd

```diff
@@ -26,13 +26,12 @@ library(gdalcubes)
 has_qgis_plugins = FALSE
 # Check to see that the qgisprocess package is installed:
 if ("qgisprocess" %in% installed.packages()) {
-  # Check that QGIS is available to prevent errors in subsequent checks:
-  if (qgisprocess::has_qgis()) {
-    # Check that the plugins are installed (they will loaded later):
-    has_grass = qgisprocess::qgis_has_provider("grass7")
-    has_saga = qgisprocess::qgis_has_provider("sagang")
-    has_qgis_plugins = has_grass && has_saga
-  }
+  # Check that QGIS and required plugins are available to prevent errors in subsequent checks:
+  has_qgis_plugins = 
+    qgisprocess::has_qgis() && 
```

Comment:
I suggest separating out the logic for QGIS being installed and for detecting plugins.

Pipes can be nice and help make data processing steps more intuitive. Here I do not find the piping helpful and suggest a simpler more verbose style would be beneficial to readers (if only readers who actually look at the source code ; )

That's my stylistic comment. In terms of the logic: we should stop instantly if QGIS is not installed. Hence previous versions had another if statement to prevent `qgis_plugins()` running when QGIS itself is not available.

Reply:
Ok, this might be a matter of taste, and yes, we can get rid of the pipe, I don't mind. But I really try to avoid unnecessary `if` conditions, thinking about it, I would find it even more elegant to write the check like this:

```r
has_qgis_plugins = 
  # make sure that qgisprocess is installed
  "qgisprocess" %in% installed.packages() &&
  # make sure that QGIS is available
  qgisprocess::has_qgis() &&
  # make sure that all required plugins are available
    all(c("grassprovider", "processing_saga_nextgen") %in% 
    qgisprocess::qgis_plugins()$name)
```

In my opinion, this is quite readable, and avoids redundancies. So if any of these conditions are unmet, qgis chunks will be not evaluated.

Reply:
I have written it like this now (but now using `qgis_has_plugin()`) but if you don't like it, feel free to change it. It is really not that important and not worth arguing about.